controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -131,25 +131,46 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	resp, err := c.OpenAppSecrets(params, nil)
 	if err != nil {
-		logger.Error(err, "Get App Secret", "appName", o.Spec.AppName)
+		logger.Error(err, "Get App Secrets", "appName", o.Spec.AppName)
+		entry, _ := r.BackOffRegistry.Get(req.NamespacedName)
+		return ctrl.Result{
+			RequeueAfter: entry.NextBackOff(),
+		}, nil
+	}
+
+	dynamicSecrets, err := getHVSDynamicSecrets(ctx, c, o.Spec.AppName)
```

Comment:
Double checking my understanding: It looks like we don't have a way currently to know if there is any dynamic/static secrets in HVS App Secrets, so VSO blindly requests HVS.

Reply:
Yeah so the `c.OpenAppSecrets()` call before this fetches all the static and rotating secrets in the HVS App. Then this function first queries the HVS App secret metadata with `c.ListAppSecrets()`, and then fetches each dynamic secret individually with `c.OpenAppSecret`. Each call to `c.OpenAppSecret` generates and returns a new set of credentials and TTL for the dynamic secret.

We're planning to cache the dynamic secret responses in the future, and then we'll know which dynamic secrets need to be fetched based on their TTLs.

Reply:
Thank you!